ODIN_II/SRC/partial_map.cpp

```diff
@@ -221,11 +221,15 @@ void partial_map_node(nnode_t* node, short traverse_number, netlist_t* netlist)
             instantiate_multi_port_mux(node, traverse_number, netlist);
             break;
         case MULTIPLY: {
-            int mult_size = std::max<int>(node->input_port_sizes[0], node->input_port_sizes[1]);
-            if (hard_multipliers && mult_size >= min_mult) {
-                instantiate_hard_multiplier(node, traverse_number, netlist);
-            } else if (!hard_adders) {
-                instantiate_simple_soft_multiplier(node, traverse_number, netlist);
+            if (mixer->softenable(operation_list::MULTIPLY)) {
+                mixer->note_candidate_node(node, operation_list::MULTIPLY);
+            } else {
+                int mult_size = std::max<int>(node->input_port_sizes[0], node->input_port_sizes[1]);
+                if (hard_multipliers && mult_size >= min_mult) {
```

Comment:
I think you still need to consider this in your optimization, so maybe this is a good candidate to be moved inside the mixingOpt class and used here, and it would validate my thought where enabled() should actually use the node

Reply:
Hi @jeanlego, it is used =) https://github.com/CAS-Atlantic/vtr-verilog-to-routing/pull/7/commits/e6a6eef79569446ff3f60a13634aa5f25025e0fa#diff-aa460a89820bf1e307c963cb7aadde78R74-R76 right here

Reply:
awesome! :) can you move that as its own function within the class and call out to it for both places? this would better organize things  and hopefully, we can move all the other types to follow this pattern later

Reply:
I will present my vision with next iteration of the code!

Reply:
Hello @jeanlego,
Thank you for your previous reviews.
The new iteration accounts with all previously discussed changes:
Since the completion of PR nears, I am asking you for another round of reviews so soon. Sorry for any inconvenience =)
Could you please review the code and suggest required changes?

Reply:
Can you make a function in mixingOpts i.e withinHardenningThreshold and use that to check?

Reply:
Do you think this https://github.com/CAS-Atlantic/vtr-verilog-to-routing/pull/7/files#diff-aa460a89820bf1e307c963cb7aadde78R74-R80 should be called from partial map?